Spring boot 4.0.4 upgrade

Seanstoppable opened the pull request:
Small upgrade to make sure we are using the latest spring boot

Seanstoppable committed:
Spring boot 3.5.6 upgrade

Small upgrade to make sure we are using the latest spring boot

masud7777r commented on build.gradle:
<img width="864" height="1920" alt="Screenshot_20251208-104356" src="https://github.com/user-attachments/assets/712aef9c-9d8f-4473-a8d2-703a0e9314b8" />

Seanstoppable committed:
Update to latest 3.5.x version

Seanstoppable committed:
Migrate tests to JUnit5

Newer versions of spring are planning to drop JUnit4 support
Newest Spring uses JUnit6
Upgrade to JUnit5 as a small jump

Seanstoppable committed:
Upgrade to newer version of gradle, for supporting newer Spring/JVM

Seanstoppable committed:
Upgrade to spring boot 4.0.2

Seanstoppable commented:
Spring Boot 3 will be EOL in a few months, so I've made the appropriate changes for Spring Boot 4 now

Seanstoppable committed:
Update to 4.0.5

Use the BOM for better dependency management

Seanstoppable committed:
Remove dependency management plugin

This was causing bootRun to fail on Spring 4
This shouldn't be necessary anymore since we are using the BOM directly,
and gradle supports it properly

Seanstoppable committed:
Upgrade to 4.0.6

Migrate to BOM from  management plugin
Normalize import format
